### ISSUE TITLE
bootMer not snowing, Windows OS

### ISSUE THREAD

**Issue by plants-22** [lme4/lme4#695]
Hi,

I think there is a bug in bootMer preventing it from working when ```parallel = "snow"```. I'm attempting to use a bootstrap over multiple CPUs in a Windows environment. Here's an example:
```r
library(glmmTMB)
library(lme4)
library(parallel)
m1 <- glmmTMB(count ~ mined + (1|site), zi=~mined,
              family=poisson, data=Salamanders)

cl <- makeCluster(6)
clusterEvalQ(cl, library("lme4"))
clusterExport(cl, varlist = c("Salamanders", "m1"))

system.time(c <- bootMer(m1, FUN = \(x)fixef(x)$cond, nsim =24, ncpus = 6, cl = cl, parallel = "snow"))

stopCluster(cl)
```
> system.time(c <- bootMer(m1, FUN = \(x)fixef(x)$cond, nsim =24, ncpus = 6, cl = cl, parallel = "snow"))
   user  system elapsed 
   0.03    0.05    0.08 
Warning message:
In bootMer(m1, FUN = function(x) fixef(x)$cond, nsim = 24, ncpus = 6,  :
  some bootstrap runs failed (24/24)

If all I change is ncpus = 1, bootMer works:
> system.time(c <- bootMer(m1, FUN = \(x)fixef(x)$cond, nsim =24, ncpus = 1, cl = cl, parallel = "snow"))
   user  system elapsed 
  21.26    0.11   21.61 

I really appreciate your time and work on lme4, any chance you could look at this problem and advise a solution?

**Comment by bbolker** [comment]
Yes, I will definitely take a look at this (I love the title!).  I'm afraid digging into this could be a bit of a nuisance as it's likely to be a problem with which environments various expressions are evaluated in, which is often hard ...

**Comment by plants-22** [comment]
Thank you!

**Comment by bbolker** [comment]
**tl;dr** you need to `clusterEvalQ(cl, library("glmmTMB"))`

I think this is a `glmmTMB`/`lme4` crossover problem;  this works (no zero-inflation because `glmer` can't handle it).

```r
m3 <- glmer(count ~ mined + (1|site),
              family=poisson, data=Salamanders)
system.time(c <- bootMer(m3, FUN = \(x)fixef(x), nsim =24, ncpus = 6, cl = cl, parallel = "snow"))
```
However, if you `clusterExport(cl, "glmmTMB")` before calling `bootMer` in your example, it works.
Here is the updated documentation for `?glmmTMB::bootmer_methods`.

Will comment further on the [glmmTMB issue](https://github.com/glmmTMB/glmmTMB/issues/843).

```r
if (requireNamespace("lme4")) {
\dontrun{
   fm1 <- glmmTMB(count~mined+(1|spp),
                  ziformula=~mined,
                  data=Salamanders,
                  family=nbinom1)
   ## single parametric bootstrap step: refit with data simulated from original model
   fm1R <- refit(fm1, simulate(fm1)[[1]])
   ## the bootMer function from lme4 provides a wrapper for doing multiple refits
   ##   with a specified summary function
   b1 <- lme4::bootMer(fm1, FUN=function(x) fixef(x)$zi, nsim=20, .progress="txt")
   if (requireNamespace("boot")) {
      boot.ci(b1,type="perc")
    }
   ## can run in parallel: may need to set up cluster explicitly,
   ## use clusterEvalQ() to load packages on workers
   if (requireNamespace("parallel")) {
      cl <- parallel::makeCluster(2)
      parallel::clusterEvalQ(cl, library("lme4"))
      parallel::clusterEvalQ(cl, library("glmmTMB"))
      b2 <- lme4::bootMer(fm1, FUN = function(x) fixef(x)$cond,
              nsim = 10, ncpus = 2, cl = cl, parallel = "snow")
   }
}
}
```

**Comment by plants-22** [comment]
Thanks Ben, much appreciated, it works!
